File: src/main/java/org/genyris/io/ParserSource.java
package org.genyris.io;

import org.genyris.core.Constants;
import org.genyris.core.Exp;
import org.genyris.core.Internable;
import org.genyris.core.PairSource;

public class ParserSource extends Parser {
    public ParserSource(Internable table, InStream stream) {
        super(table, stream, Constants.DYNAMICSCOPECHAR2, Constants.CDRCHAR,
                Constants.COMMENTCHAR);
    }

    public ParserSource(Internable table, InStream stream, char dynaChar,
            char cdrCharacter, char commentChar) {
        super(table, stream, dynaChar, cdrCharacter, commentChar);
    }
    @Override
    protected Exp cons(Exp l, Exp r, int line) {
        Exp retval = new PairSource(l, r, _lexer.getFilename(), line);
        return retval;
    }

}

File: src/org/genyris/io/ParserSource.kt
package org.genyris.io

import org.genyris.core.Constants
import org.genyris.core.Exp
import org.genyris.core.Internable
import org.genyris.core.PairSource

class ParserSource : Parser {
    constructor(table: Internable?, stream: InStream?) : super(
        table, stream, Constants.DYNAMICSCOPECHAR2, Constants.CDRCHAR,
        Constants.COMMENTCHAR
    )

    constructor(
        table: Internable?, stream: InStream?, dynaChar: Char,
        cdrCharacter: Char, commentChar: Char
    ) : super(table, stream, dynaChar, cdrCharacter, commentChar)

    override fun cons(l: Exp?, r: Exp?, line: Int): Exp {
        val retval: Exp = PairSource(l, r, _lexer.getFilename(), line)
        return retval
    }
}
